### PR TITLE
Dt 1268 search by team code

### PR DESCRIPTION
**Pull request by rgforsyth**
A new endpoint to allow search by team code. This will return all offenders who have an active manager that is part of a team containing a given team code.

Currently the search service code for search by team code & search by ldu are very similar. This can be changed soon when the /ldu-codes endpoint implements paging.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchTeamCodeListAPIIntegrationTest.kt [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+class OffenderSearchTeamCodeListAPIIntegrationTest : LocalstackIntegrationBase() {
+
+    @BeforeEach
+    fun setUp() {
+        loadOffenders(
+                OffenderReplacement(
+                        offenderManagers = listOf(
+                                OffenderManagerReplacement(
+                                        team = TeamReplacement(code="N01000"
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchTeamCodeListAPIIntegrationTest.kt:18]: Ooh this level of indentation is begging for some ktlint :)

File: src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchTeamCodeListAPIIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .`as`(Array<OffenderDetail>::class.java)
+
+        Assertions.assertThat(results).hasSize(3)
+        Assertions.assertThat(results[0].offenderManagers?.get(0)?.team?.code).contains("N02000")
+        Assertions.assertThat(results[1].offenderManagers?.get(0)?.team?.code).contains("N01000")
+        Assertions.assertThat(results[1].offenderManagers?.get(0)?.team?.code).contains("N01000")
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchTeamCodeListAPIIntegrationTest.kt:152]: extra line here can be removed

Comment (mikehalmamoj) [reply src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchTeamCodeListAPIIntegrationTest.kt:152]: whoops, no sorry, should be checking results[2]?

Comment (rgforsyth) [reply src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchTeamCodeListAPIIntegrationTest.kt:152]: It should!

**Review by mikehalmamoj — APPROVED** [review]

**Review by andymarke — COMMENTED** [review]
LGTM

**Pull request merged**